code/modules/reagents/chemistry/reagents/blob_reagents.dm

```diff
@@ -135,11 +143,13 @@
 
 /datum/reagent/blob/hallucinogenic_nectar/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
 	reac_volume = ..()
-	M.apply_damage(0.4*reac_volume, TOX)
-	M.hallucination += 0.6*reac_volume
-	M.druggy += 0.6*reac_volume
-	if(M.reagents)
-		M.reagents.add_reagent("spore", 0.2*reac_volume)
+	if(M)
+		M.apply_damage(0.4*reac_volume, TOX)
+	if(M)
+		M.hallucination += 0.6*reac_volume
```

Comment:
if the above can destroy the mob, why not do it AFTER this.
surely that would avoid ```if(M) if(M)```